### PR TITLE
[WNMGDS-92] Add react badge component

### PR DESCRIPTION
**Pull request by bernardwang**
### Added
<Badge> react component and updated react examples.

### CODE REVIEW

File: packages/core/src/components/Badge/Badge.example.jsx [hunk 1]
```diff
@@ -5,12 +5,13 @@ import ReactDOM from 'react-dom';
 ReactDOM.render(
   <Fragment>
     <Badge>Default badge</Badge>
-    <Badge variation="info">Variant badge</Badge>
+    <Badge variation="info">Info badge</Badge>
     <Badge variation="success">
-      Badge with <b>nested</b>
+      Success Badge with <b>bold text</b>
```
Comment (bernardwang) [inline packages/core/src/components/Badge/Badge.example.jsx]: Hm, the only thought I have is that this is supposed to demonstrate an example with more complex `children` or multiple nested elements. Whether its bold or something else doesnt actually matter

File: packages/core/src/components/Badge/Badge.example.jsx [hunk 2]
```diff
@@ -5,12 +5,13 @@ import ReactDOM from 'react-dom';
 ReactDOM.render(
   <Fragment>
     <Badge>Default badge</Badge>
-    <Badge variation="info">Variant badge</Badge>
+    <Badge variation="info">Info badge</Badge>
```
Comment (bernardwang) [inline packages/core/src/components/Badge/Badge.example.jsx]: Same with this, I would say this example is demonstrating the `variation` prop. Maybe something like "Badge with `variation` prop"?

File: packages/core/src/components/Badge/Badge.jsx [hunk 3]
```diff
@@ -17,7 +17,7 @@ export const Badge = props => {
 
 Badge.propTypes = {
   /**
-   * Additional classes to be added to the root button element.
```
Comment (bernardwang) [inline packages/core/src/components/Badge/Badge.jsx]: Nice catch!

**Comment by bernardwang** [comment]
@line47 Referring to the naming, the most important thing for me is that the naming describes the case/functionality we want to highlight. In the HTML examples this was pretty straightforward because you want to show all the `classes`. However in the React examples, I feel its more important to showcase the React props and the different ways to use the React component. So that's why I chose the names that I did, in order to describe the React specific use cases rather than the styles, which is already covered in the HTML examples. What do you think about that?

File: packages/core/src/components/Badge/Badge.example.jsx (hunk 1 above)
Comment (line47) [reply packages/core/src/components/Badge/Badge.example.jsx]: Ah, I see what you are saying. Makes sense to leave it as is

**Comment by bernardwang** [comment]
@line47 If you want to take a look again, I added a size prop and updated the examples!

File: packages/core/src/components/Badge/Badge.jsx [hunk 4]
```diff
@@ -1,16 +1,40 @@
...
    * In most cases this will be the badge's label, but you could also use this
    * to nest more advanced JSX.
-  */
-  children: PropTypes.oneOfType([PropTypes.node, PropTypes.string]).isRequired
+   */
+  children: PropTypes.oneOfType([PropTypes.node, PropTypes.string]).isRequired,
+  /**
+   * Sets the font size of the Badge
+   */
+  size: PropTypes.oneOfType(['big']),
```
Comment (line47) [inline packages/core/src/components/Badge/Badge.jsx]: I think the size propType should be `oneOf`. Right now the prop table shows `custom` where I think it should be showing `big` similar to the way the button component props are set up.
Screen Shot 2019-11-12 at 10 04 21 AM

Comment (bernardwang) [reply packages/core/src/components/Badge/Badge.jsx]: thanks for the catch!

**Pull request merged**